### PR TITLE
#163620127 GET /parties/:id

### PR DESCRIPTION
**Pull request by frankly034**
#### What does this PR do?
Have api endpoint for GET /parties/:id

#### Description of Task to be completed?
* Create party model function to fetches a specific party from the resource
* Create party controller function to manage fetch specific party requests
* Create party route to handle GET /parties/:id request
* Create parties/:id test
- ensure non-numeric id return status: 400
- ensure empty search to return status: 404
- return 200 and data if found

#### How should this be manually tested?
After cloning the repo, run 'npm run test'

#### Any background context you want to provide?
* Http test were done using supertest and chai expect library

#### What are the relevant pivotal tracker stories?
#163620127

### CODE REVIEW

**Comment by frankly034** [comment]
@Lumexralph guess your having a good day. Here's another PR

File: server/middlewares/partyMiddleware.js [hunk 1]
```diff
@@ -7,6 +7,15 @@ class PartyMiddleware {
     }
     return next();
   }
+
+  static getAParty(req, res, next) {
+    const { id } = req.params;
+    if (isNaN(parseInt(id, 10))) {
```
Comment (Lumexralph) [inline server/middlewares/partyMiddleware.js]: @frankly034 Good work so far! please use the `Number.isNaN` as against the `isNaN` global method you used

Comment (frankly034) [reply server/middlewares/partyMiddleware.js]: Thank you very much. Was thinking what that eslint error meant.

Comment (frankly034) [reply server/middlewares/partyMiddleware.js]: Resolved some conflicts.

**Review by Lumexralph — APPROVED** [review]
Good Job! This work looks good to me.

**Pull request merged**